pandas/core/dtypes/cast.py

```diff
@@ -1221,8 +1221,13 @@ def construct_1d_arraylike_from_scalar(value, length, dtype):
 
         # coerce if we have nan for an integer dtype
         if is_integer_dtype(dtype) and isna(value):
-            dtype = np.float64
-        subarr = np.empty(length, dtype=dtype)
+            dtype = np.dtype('float64')
+        if isinstance(dtype, np.dtype) and dtype.kind in ("U", "S"):
```

Comment:
I think `dtype.kind in ("U", "S")` can be replaced with `is_string_dtype(dtype)`.

`is_string_dtype` can be found in `pandas/core/dtypes/common`

Reply:
this can also be an elif